ninja/src/main/java/controllers/ApplicationController.java

```diff
@@ -0,0 +1,102 @@
+package controllers;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import com.google.inject.Provider;
+import com.google.inject.Singleton;
+import com.google.inject.persist.Transactional;
+
+import models.User;
+import ninja.Context;
+import ninja.Result;
+import ninja.Results;
+import ninja.i18n.Lang;
+import ninja.i18n.Messages;
+import ninja.jpa.UnitOfWork;
+import ninja.session.FlashScope;
+import ninja.validation.JSR303Validation;
+import ninja.validation.Validation;
+import services.UserService;
+
+@Singleton
+public class ApplicationController {
+    
+    @Inject
+    Lang lang;
+
+    @Inject
+    Messages msg;
+    
+    private static Log logger = LogFactory.getLog(ApplicationController.class);
+    
+    @Inject 
+    Provider<EntityManager> entitiyManagerProvider;
```

Comment:
rename this to entityManagerProvider (there's an extra 'i' in the name)

Reply:
typo fixed